src/createRollupPrepareDeploymentParamsConfigDefaults.ts

```diff
@@ -29,3 +38,22 @@ export const defaults = {
   numBigStepLevel: 1,
   wasmModuleRoot: getConsensusReleaseByVersion(32).wasmModuleRoot,
 } as const;
+
+export function createRollupPrepareDeploymentParamsConfigDefaults(
+  rollupCreatorVersion: 'v2.1',
+): typeof defaultsV2Dot1;
+export function createRollupPrepareDeploymentParamsConfigDefaults(
+  rollupCreatorVersion: 'v3.1',
+): typeof defaultsV3Dot1;
+export function createRollupPrepareDeploymentParamsConfigDefaults(
+  rollupCreatorVersion?: never,
+): typeof defaultsV3Dot1;
+export function createRollupPrepareDeploymentParamsConfigDefaults(
+  rollupCreatorVersion: 'v3.1' | 'v2.1' = 'v3.1',
+) {
```

Comment:
made it into a function so we can version it + added overloads so return type is based on provided version